prusti-viper/src/encoder/encoder.rs

```diff
@@ -1075,6 +1076,8 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
             ty::TyKind::Uint(ty::UintTy::U64) => scalar_value.to_u64().unwrap().into(),
             ty::TyKind::Uint(ty::UintTy::U128) => scalar_value.to_u128().unwrap().into(),
             ty::TyKind::Uint(ty::UintTy::Usize) => scalar_value.to_machine_usize(&self.env().tcx()).unwrap().into(),
+            ty::TyKind::Float(ty::FloatTy::F32) => unimplemented!("to_f32 not yet implemented"),
+            ty::TyKind::Float(ty::FloatTy::F64) => scalar_value.to_f64().unwrap().into(),
```

Comment:
@vakaras Compilation fails here with error message

> error[E0277]: the trait bound `prusti_common::vir::Expr: From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::DoubleS>>` is not satisfied
    --> prusti-viper/src/encoder/encoder.rs:1080:83
     |
1080 |             ty::TyKind::Float(ty::FloatTy::F64) => scalar_value.to_f64().unwrap().into(),
     |                                                                                                                            ^^^^ the trait `From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::DoubleS>>` is not implemented for `prusti_common::vir::Expr`
     |
     = help: the following implementations were found:
               <prusti_common::vir::Expr as From<&'a LocalVar>>
               <prusti_common::vir::Expr as From<&'a str>>
               <prusti_common::vir::Expr as From<LocalVar>>
               <prusti_common::vir::Expr as From<bool>>
             and 16 others

However, among the '16 other' implementations is `<prusti_common::vir::Expr as From<rustc_apfloat::ieee::IeeeFloat<rustc_apfloat::ieee::DoubleS>>>` in [prusti-common/src/vir/conversions.rs](https://github.com/viperproject/prusti-dev/pull/451/files#diff-b34dcdcd8d6eeb45501fe2f693881bb0658cfe1899099e091436e845b1678ccf).

So I am at a loss as to what goes wrong.